### PR TITLE
Open file limit not set correctly

### PR DESCRIPTION
**Pull request by jessedavis**
If node['rabbitmq']['open_file_limit'] is set to anything other than the default, the resulting ulimit line is normally written in /etc/rabbitmq/rabbitmq-env.conf.  That file is usually sourced, and doing so usually prints out an error along the lines of "operation not permitted" when using rabbitmqctl, etc.

The correct place for the ulimit is in /etc/default/rabbitmq-server.  I tested this manually by applying the updated recipe to the default-centos-65 and default-ubuntu-1204 images.

I followed the same format for the minitest tests, but could not figure out how to execute them.  I'd appreciate a quick rundown on how that's supposed to work, as I'm pretty new to Chef and test-kitchen and am just learning them now.

### CODE REVIEW

**Comment by jessedavis** [comment]
@jjasghar I added a spec (my first time using ChefSpec), so please feel free to reject if I did it wrong.  I did have to upgrade ChefSpec above 4.0 to get the tests to start.  Since this is my first time contributing to and using Chef, is it normal best practice to perform `bundle upgrade` frequently to pull in updated dependencies?

File: recipes/default.rb [hunk 1]
```diff
@@ -184,6 +184,14 @@ class Chef::Resource # rubocop:disable all
   notifies :restart, "service[#{node['rabbitmq']['service_name']}]", :immediately
 end
 
+template "/etc/default/#{node['rabbitmq']['service_name']}" do
+  source 'default.rabbitmq-server.erb'
+  owner 'root'
+  group 'root'
+  mode 00644
```
Comment (tas50) [inline recipes/default.rb]: This should be a 4 digit string: '0644'

File: spec/default_spec.rb [hunk 2]
```diff
@@ -47,6 +47,15 @@
     expect(chef_run).to install_package('erlang')
   end
 
+  it 'should create the rabbitmq /etc/default file' do
+    expect(chef_run).to create_template("/etc/default/#{chef_run.node['rabbitmq']['service_name']}").with(
+      :user => 'root',
+      :group => 'root',
+      :source => 'default.rabbitmq-server.erb',
+      :mode => 00644
```
Comment (tas50) [inline spec/default_spec.rb]: This one as well

**Comment by stevecstian** [comment]
@tas50 It should be fine to specify 5 digit integer for the value of mode attribute as we're using that format in default recipe.

File: Gemfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ end
 
 group :unit do
   gem 'berkshelf'
-  gem 'chefspec'
+  gem 'chefspec', '~> 4.0'
```
Comment (stevecstian) [inline Gemfile]: @jjasghar Is this required?

Comment (stevecstian) [inline Gemfile]: Paging @jjasghar 
Will Gemfile.lock help on this?
/CC @michaelklishin

Comment (jjasghar) [inline Gemfile]: Meh, you should be running the newest version of chefspec right?

**Comment by jessedavis** [comment]
@jjasghar I think this PR should be ready for a merge (sorry about the long delay).

**Comment by jjasghar** [comment]
@jessedavis we've made some comments on the PR, can you either respond or update the PR with those changes? Thanks!

File: recipes/default.rb (hunk 1 above)
Comment (cmluciano) [inline recipes/default.rb]: The 5 digit should be fine. I believe this just forces the octet rule

File: spec/default_spec.rb (hunk 2 above)
Comment (cmluciano) [inline spec/default_spec.rb]: The 5 digit should be fine. I believe this just forces the octet rule

**Comment by jessedavis** [comment]
@jjasghar I believe I've made all the changes that you and @stevecstian recommended in the comments.  Let me know if I'm mistaken or if I didn't make the changes correctly.

**Comment by johnbellone** [comment]
:+1:

**Comment by ikurochkin** [comment]
:+1:

**Pull request merged**